examples/crack_branching.cpp

```diff
@@ -18,134 +18,137 @@
+    double thickness = inputs["system_size"][2];
+    double L_prenotch = height / 2.0;
+    double y_prenotch1 = 0.0;
+    Kokkos::Array<double, 3> p01 = { low_corner[0], y_prenotch1,
+                                     low_corner[2] };
+    Kokkos::Array<double, 3> v1 = { L_prenotch, 0, 0 };
+    Kokkos::Array<double, 3> v2 = { 0, 0, thickness };
+    Kokkos::Array<Kokkos::Array<double, 3>, 1> notch_positions = { p01 };
+    CabanaPD::Prenotch<1> prenotch( v1, v2, notch_positions );
+
+    // ====================================================
+    //                    Force model
+    // ====================================================
+    using model_type = CabanaPD::ForceModel<CabanaPD::PMB, CabanaPD::Fracture>;
+    model_type force_model( delta, K, G0 );
+
+    // ====================================================
+    //                 Particle generation
+    // ====================================================
+    // Does not set displacements, velocities, etc.
+    auto particles = std::make_shared<
+        CabanaPD::Particles<memory_space, typename model_type::base_model>>(
+        exec_space(), low_corner, high_corner, num_cells, halo_width );
+
+    // ====================================================
+    //                Boundary conditions
+    // ====================================================
+    double sigma0 = inputs["traction"];
+    double dy = particles->dx[1];
+    double b0 = sigma0 / dy;
+
+    CabanaPD::RegionBoundary plane1( low_corner[0], high_corner[0],
+                                     low_corner[1] - dy, low_corner[1] + dy,
+                                     low_corner[2], high_corner[2] );
+    CabanaPD::RegionBoundary plane2( low_corner[0], high_corner[0],
+                                     high_corner[1] - dy, high_corner[1] + dy,
+                                     low_corner[2], high_corner[2] );
+    std::vector<CabanaPD::RegionBoundary> planes = { plane1, plane2 };
+    auto particles_f = particles->getForce();
+    auto particles_x = particles->getReferencePosition();
+    // Create a symmetric force BC in the y-direction.
+    auto bc_op = KOKKOS_LAMBDA( const int pid )
+    {
+        // Get a modifiable copy of force.
+        auto p_f = particles_f.getParticleView( pid );
+        // Get a copy of the position.
+        auto p_x = particles_x.getParticle( pid );
+        auto ypos = Cabana::get( p_x, CabanaPD::Field::ReferencePosition(), 1 );
+        auto sign = std::abs( ypos ) / ypos;
+        Cabana::get( p_f, CabanaPD::Field::Force(), 1 ) += b0 * sign;
+    };
+    auto bc =
+        createBoundaryCondition( bc_op, exec_space{}, *particles, planes );
+
+    // ====================================================
+    //            Custom particle initialization
+    // ====================================================
+    auto rho = particles->sliceDensity();
+    auto x = particles->sliceReferencePosition();
+    auto v = particles->sliceVelocity();
```

Comment:
@streeve Do we need lines 120 and 121? We are not using "v" or "f" within init_functor.

Reply:
Can you open a branch for this separately?